http_service/bugbug_http/app.py

```diff
@@ -85,24 +91,73 @@ class UnauthorizedError(Schema):
     message = fields.Str(default="Error, missing X-API-KEY")
 
 
+class BranchName(Schema):
+    branch = fields.Str(example="autoland")
+
+
+class Schedules(Schema):
+    tasks = fields.List(fields.Str)
+    groups = fields.List(fields.Str)
+
+
+class PushNotFoundError(Schema):
+    message = fields.Str(default="Error: specified push not found!")
+
+
 spec.components.schema(BugPrediction.__name__, schema=BugPrediction)
-spec.components.schema(
-    BugPredictionNotAvailableYet.__name__, schema=BugPredictionNotAvailableYet
-)
+spec.components.schema(NotAvailableYet.__name__, schema=NotAvailableYet)
 spec.components.schema(ModelName.__name__, schema=ModelName)
 spec.components.schema(UnauthorizedError.__name__, schema=UnauthorizedError)
+spec.components.schema(BranchName.__name__, schema=BranchName)
+spec.components.schema(Schedules.__name__, schema=Schedules)
+spec.components.schema(PushNotFoundError.__name__, schema=PushNotFoundError)
 
 
 api_key_scheme = {"type": "apiKey", "in": "header", "name": "X-API-Key"}
 spec.components.security_scheme("api_key", api_key_scheme)
 
 
+def get_target_func():
+    if "predict" in request.path:
+        return classify_bug
+
+    if request.path.startswith("/push"):
+        return schedule_tests
+
+    raise ValueError(f"Invalid request: {request.path}")
+
+
 def get_job_id():
     return uuid.uuid4().hex
 
 
-def get_mapping_key(model_name, bug_id):
-    return f"bugbug:mapping_{model_name}_{bug_id}"
+def get_mapping_key(*args):
+    """Return a key that can be used to map `args` to a job id.
+
+    If args aren't specified, a key that maps `request.path` to a job id will
+    be used instead.
+    """
+    func = get_target_func()
+    return f"bugbug:job_id:{func.__name__}:{'_'.join(args)}"
+
+
+def get_result_key(*args, func=None):
+    """Return a key that can be used to store the result of a job.
+
+    If args aren't specified, a key that maps `request.path` to a job result
+    will be used instead.
+    """
+    func = func or get_target_func()
+    return f"bugbug:job_result:{func.__name__}:{'_'.join(args)}"
+
+
+def schedule_job(*args, job_id=None):
```

Comment:
And here I'd make `func` a parameter of `schedule_job`, so we can get rid of get_target_func and make it more explicit

Reply:
Will do, I was trying to avoid needing to pass it around everywhere.. but agree it ended up being more fragile than it needed to.